dll/appcompat/shims/genral/timeapi.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * PROJECT:     ReactOS 'General' Shim library
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     TimeAPI related shims
+ * COPYRIGHT:   Copyright 2016,2017 Mark Jansen (mark.jansen@reactos.org)
+ */
+
+#define WIN32_NO_STATUS
+#include <windef.h>
+#include <winbase.h>
+#include <winuser.h>
+#include <wingdi.h>
+#include <shimlib.h>
+#include <strsafe.h>
+#include <mmsystem.h>
+
+#define SHIM_NS         DefaultTimerResolution1ms
+#include <setup_shim.inl>
+
+typedef DWORD (WINAPI* TIMEGETTIMEPROC)(void);
+typedef MMRESULT(WINAPI* TIMEBEGINPERIODPROC)(_In_ UINT);
+typedef MMRESULT(WINAPI* TIMEENDPERIODPROC)(_In_ UINT);
+
+/* Only adjust timer resolution if timeGetTime() is called before timeBeginPeriod() */
+static LONG isResolutionSet;
+
+#define SHIM_NOTIFY_FN SHIM_OBJ_NAME(Notify)
+
+BOOL WINAPI SHIM_OBJ_NAME(Notify)(DWORD fdwReason, PVOID ptr)
+{
+    HMODULE hWinMM;
+    TIMEENDPERIODPROC ftimeEndPeriod;
+    
+    if (fdwReason == SHIM_REASON_DEINIT)
+    {
+        hWinMM = GetModuleHandleW(L"WINMM");
+        if (hWinMM != NULL)
+        {
+            ftimeEndPeriod  = (TIMEENDPERIODPROC)GetProcAddress(hWinMM, "timeEndPeriod");
+            if (ftimeEndPeriod != NULL && InterlockedAnd(&isResolutionSet, TRUE))
```

Comment:
You also 'end' it here when you did not set it yourself (`APIHook_timeBeginPeriod` sets the same thing)